src/main/java/com/faforever/api/user/UserService.java

```diff
@@ -423,6 +426,40 @@ public CallbackResult linkToSteam(String token, String steamId) {
     return new CallbackResult(callbackUrl, errors);
   }
 
+  public String buildGogToken(User user) {
+    return String.format(properties.getGog().getTokenFormat(), user.getId());
+  }
+
+  public void linkGogAccount(String gogUsername, User user) { // TODO: rate limit?
```

Comment:
Any ideas on rate limiting so we don't send lots of external requests?

Reply:
This method should be @Transactional.

To your question: No, I would not rate limit.

Reply:
ah, right, that was it, I remembered something, but wasn't sure what it was. Also, sry @transactional